qiskit_experiments/library/characterization/cr_hamiltonian_analysis.py

```diff
@@ -0,0 +1,294 @@
+                x, px=px1, py=py1, pz=pz1, baseline=b
+            ),
+            filter_kwargs={"control_state": 1, "meas_basis": "x"},
+            plot_color="red",
+            plot_symbol="^",
+            canvas=0,
+        ),
+        curve.SeriesDef(
+            name="y|c=1",
+            fit_func=lambda x, px0, px1, py0, py1, pz0, pz1, b: curve.fit_function.bloch_oscillation_y(
+                x, px=px1, py=py1, pz=pz1, baseline=b
+            ),
+            filter_kwargs={"control_state": 1, "meas_basis": "y"},
+            plot_color="red",
+            plot_symbol="^",
+            canvas=1,
+        ),
+        curve.SeriesDef(
+            name="z|c=1",
+            fit_func=lambda x, px0, px1, py0, py1, pz0, pz1, b: curve.fit_function.bloch_oscillation_z(
+                x, px=px1, py=py1, pz=pz1, baseline=b
+            ),
+            filter_kwargs={"control_state": 1, "meas_basis": "z"},
+            plot_color="red",
+            plot_symbol="^",
+            canvas=2,
+        ),
+    ]
+
+    @classmethod
+    def _default_options(cls):
+        """Return the default analysis options."""
+        default_options = super()._default_options()
+        default_options.data_processor = dp.DataProcessor(
+            input_key="counts", data_actions=[dp.Probability("1"), dp.BasisExpectationValue()]
+        )
+        default_options.curve_plotter = "mpl_multiv_canvas"
+        default_options.xlabel = "Net cross resonance duration"
+        default_options.ylabel = "<X(t)>,<Y(t)>,<Z(t)>"
+        default_options.xval_unit = "s"
+        default_options.style = curve.visualization.PlotterStyle(
+            figsize=(8, 10),
+            legend_loc="lower right",
+            fit_report_rpos=(0.28, -0.10),
+        )
+        default_options.ylim = (-1, 1)
+
+        return default_options
+
+    def _setup_fitting(self, **extra_options) -> Union[Dict[str, Any], List[Dict[str, Any]]]:
+        """Fitter options."""
+        user_p0 = self._get_option("p0")
+        user_bounds = self._get_option("bounds")
+
+        bounds = {
+            "px0": user_bounds["px0"] or (-np.inf, np.inf),
+            "py0": user_bounds["py0"] or (-np.inf, np.inf),
+            "pz0": user_bounds["pz0"] or (-np.inf, np.inf),
+            "px1": user_bounds["px1"] or (-np.inf, np.inf),
+            "py1": user_bounds["py1"] or (-np.inf, np.inf),
+            "pz1": user_bounds["pz1"] or (-np.inf, np.inf),
+            "b": user_bounds["b"] or (-1, 1),
+        }
+
+        guesses = defaultdict(list)
+
+        for control in (0, 1):
+            # start from Z oscillation
+            x_data = self._data(series_name=f"x|c={control}")
+            y_data = self._data(series_name=f"y|c={control}")
+            z_data = self._data(series_name=f"z|c={control}")
+
+            omega_xyz = []
+            for data in (x_data, y_data, z_data):
+                fft_freq = curve.guess.frequency(data.x, data.y)
```

Comment:
In our old code, there are extra guesses using linear fits to the data to handle the low frequency case. Have you tested with low frequency data? (Most relevant when measuring cross-talk and stray coupling).